packages/toolpad-core/src/AppProvider/AppProvider.tsx

```diff
@@ -1,23 +1,52 @@
 import * as React from 'react';
-import { AppRouterCacheProvider } from '@mui/material-nextjs/v14-appRouter';
 import { ThemeProvider } from '@mui/material/styles';
 import CssBaseline from '@mui/material/CssBaseline';
-
 import type { Theme } from '@emotion/react';
 
+export interface Branding {
+  title?: string;
+  logo?: React.ReactNode;
+}
+
+export interface NavigationPageItem {
+  kind?: 'page';
+  title: string;
+  path?: string;
+  icon?: React.ReactNode;
+  children?: Navigation;
+}
+
+export interface NavigationSubheaderItem {
+  kind: 'header';
+  title: string;
+}
+
+export interface NavigationDividerItem {
+  kind: 'divider';
+}
+
+export type NavigationItem = NavigationPageItem | NavigationSubheaderItem | NavigationDividerItem;
+
+export type Navigation = NavigationItem[];
+
+export const BrandingContext = React.createContext<Branding | null>(null);
+
+export const NavigationContext = React.createContext<Navigation>([]);
+
 interface AppProviderProps {
   children: React.ReactNode;
   theme: Theme;
+  branding?: Branding | null;
+  navigation: Navigation;
```

Comment:
Default navigation can be `[]`? i.e. I should be allowed to drop in a dashboard layout and concentrate on the refining the branding before adding pages. Or vice versa.

```suggestion
  navigation?: Navigation;
```